src/pd/client.rs

```diff
@@ -72,17 +76,81 @@ fn parse_urls(addr: &str) -> Result<Vec<String>> {
     Ok(hosts)
 }
 
-fn rpc_connect(addr: &str) -> Result<TcpStream> {
-    let hosts = try!(parse_urls(addr));
+fn parse_pd_members(json: &Json) -> Result<Vec<String>> {
+    let mut hosts = Vec::new();
+
+    if let Some(&Json::Array(ref members)) = json.find("members") {
+        for m in members {
+            if let Some(&Json::Array(ref client_urls)) = m.find("client-urls") {
+                for url in client_urls {
+                    if let &Json::String(ref u) = url {
+                        hosts.extend(try!(parse_urls(u)))
+                    }
+                }
+            }
+        }
+    }
+
+    if hosts.len() > 0 {
+        Ok(hosts)
+    } else {
+        Err(box_err!("invalid pd members object"))
+    }
+}
+
+fn get_pd_members(endpoints: &str) -> Result<Vec<String>> {
```

Comment:
We can get rid of hyper if we don't need to get leader, and the implementation will be simpler.